cores/arduino/stm32/analog.c

```diff
@@ -110,10 +112,12 @@
   * @}
   */
 
-/** @addtogroup STM32F4xx_System_Private_Variables
-  * @{
-  */
+ADC_HandleTypeDef AdcHandleTab[ADC_NUM];
+static DMA_HandleTypeDef DmaHandleTab[ADC_NUM] = {NULL};
+static PinName g_current_pinTab[ADC_NUM];
 static PinName g_current_pin = NC;
```

Comment:
looks like g_current_pin  is not used anymore right ?

Reply:
It's used for DAC function, I don't touch to this part. See for example function HAL_DAC_MspInit() or dac_write_value().

Reply:
hmm ok - can we use the same DAC in DMA then without DMA ? Isn't it supposed to be the same pin .

Reply:
Do you want to use g_current_pinTab for DAC too? I prefer to leave as it is because DAC and ADC are two different peripherals.

Reply:
Ok - no please keep them separated if they're used for different purpose. But would be great to have at a least a comment to explain that one is used for ADC and the other one is used for DAC because as of now they have exactly the same name.